hera/editor/MapEditor.tsx

```diff
@@ -212,7 +261,8 @@ export default function MapEditor({
     (size = new SizeVector(random(10, 15), random(10, 15))): MapData => {
       return withHumanPlayer(
         mapObject
-          ? MapData.fromJSON(mapObject.state)!
+          ? getInitialMapFromUndoStack(mapObject?.id)?.map ??
```

Comment:
this restores the state from the correct position in the undostack when loading the map from the url